server/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentGenerator.java

```diff
@@ -316,7 +316,7 @@ public void writeRawField(String name, InputStream content, XContentType content
         if (mayWriteRawData(contentType) == false) {
             // EMPTY is safe here because we never call namedObject when writing raw data
             try (XContentParser parser = XContentFactory.xContent(contentType)
-                    .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, content)) {
+                    .createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION, content)) {
```

Comment:
I think an explanation why it's ok to throw an exception here might be helpful for future us.

Reply:
Sure, I'll add a comment, good idea